core/iwasm/common/wasm_application.c

```diff
@@ -250,6 +253,11 @@ wasm_application_execute_main(WASMModuleInstanceCommon *module_inst, int32 argc,
     }
 #endif
 
+#if WASM_ENABLE_GC_PERF_PROFILING != 0
+    void *handle = wasm_runtime_get_gc_heap_handle(module_inst);
+    gc_dump_perf_profiling(handle);
```

Comment:
Had better add an API in `core/shared/mem-alloc/mem_alloc.h/c`, for example `mem_allocator_dump_perf_profiling`, which calls `gc_dump_perf_profiling`. We may use different GC allocators in the future, not only ems, and normally runtime only calls API in mem_allocator.